spec/forms/backup_code_verification_form_spec.rb

```diff
@@ -0,0 +1,45 @@
+require 'rails_helper'
+
+RSpec.describe BackupCodeVerificationForm do
+  let(:form) { described_class.new(user) }
+  let(:user) { create(:user, :fully_registered) }
+  let(:backup_codes) do
+    BackupCodeGenerator.new(user).create
+  end
+  let(:params) do
+    {
+      backup_code: code,
+    }
+  end
+  subject(:result) { form.submit(params) }
+
+  describe '#submit' do
+    context 'with a valid backup code' do
+      let(:code) { backup_codes.first }
+      let(:backup_code_config) do
+        BackupCodeConfiguration.find_with_code(code: code, user_id: user.id)
+      end
+
+      it 'returns succcess' do
+        expect(result.success?).to eq(true)
+        expect(result.extra[:multi_factor_auth_method]).to eq('backup_code')
+        expect(result.extra[:multi_factor_auth_method_created_at]).
+          to eq(backup_code_config.created_at)
```

Comment:
I think we could be proactive to force a future maintainer to update these specs with any new revisions by asserting equality, so that any new keys would fail the spec.

Either:

```suggestion
        expect(result.success?).to eq(true)
        expect(result.extra).to eq(
          multi_factor_auth_method: 'backup_code',
          multi_factor_auth_method_created_at: backup_code_config.created_at,
        )
```

Or:

```suggestion
        expect(result.to_h).to eq(
          success: true,
          errors: {},
          multi_factor_auth_method: 'backup_code',
          multi_factor_auth_method_created_at: backup_code_config.created_at,
        )
```